shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/BidsImporterService.java

```diff
@@ -0,0 +1,241 @@
+	 * @throws IOException 
+	 */
+	private void importDataset(ImportJob importJob, BidsDataType bidsDataType, DatasetModalityType modalityType, ShanoirEvent event) throws IOException {
+
+		DatasetAcquisition datasetAcquisition = new BidsDatasetAcquisition();
+
+		// Get examination
+		Examination examination = examinationRepository.findById(importJob.getExaminationId()).get();
+		datasetAcquisition.setExamination(examination);
+
+		List<Dataset> datasets = new ArrayList<>();
+		float progress = 0f;
+
+		File[] filesToImport = new File(importJob.getWorkFolder()).listFiles();
+
+		for (File importedFile : filesToImport) {
+			progress += 1f / filesToImport.length;
+			event.setMessage("Bids dataset for examination " + importJob.getExaminationId());
+			event.setProgress(progress);
+			eventService.publishEvent(event);
+
+			// Metadata
+			DatasetMetadata originMetadata = new DatasetMetadata();
+			originMetadata.setDatasetModalityType(modalityType);
+			originMetadata.setCardinalityOfRelatedSubjects(CardinalityOfRelatedSubjects.SINGLE_SUBJECT_DATASET);
+
+			// Create the dataset with informations from job
+			BidsDataset datasetToCreate = new BidsDataset();
+			datasetToCreate.setSubjectId(examination.getSubjectId());
+
+			// DatasetExpression with list of files
+			DatasetExpression expression = new DatasetExpression();
+			expression.setCreationDate(LocalDateTime.now());
+			expression.setDatasetExpressionFormat(DatasetExpressionFormat.BIDS);
+			expression.setDataset(datasetToCreate);
+
+			List<DatasetFile> files = new ArrayList<>();
+
+			// Copy the data to "BIDS" folder
+			final String subLabel = SUBJECT_PREFIX + importJob.getSubjectName();
+			final String sesLabel = SESSION_PREFIX + importJob.getExaminationId();
+
+			final File outDir = new File(niftiStorageDir + File.separator + File.separator + subLabel + File.separator + sesLabel + File.separator + bidsDataType.getFolderName());
+			outDir.mkdirs();
+			String outPath = outDir.getAbsolutePath() + File.separator + importedFile.getName();
+			Path importedFileFinalLocation = Files.copy(importedFile.toPath(), Paths.get(outPath), StandardCopyOption.REPLACE_EXISTING);
+
+			DatasetFile dsFile = new DatasetFile();
+			dsFile.setDatasetExpression(expression);
+			dsFile.setPacs(false);
+			dsFile.setPath(importedFileFinalLocation.toUri().toString().replaceAll(" ", "%20"));
+			files.add(dsFile);
+
+			expression.setDatasetFiles(files);
+			datasetToCreate.setDatasetExpressions(Collections.singletonList(expression));
+
+			// Fill dataset with informations
+			datasetToCreate.setCreationDate(LocalDate.now());
+			datasetToCreate.setDatasetAcquisition(datasetAcquisition);
+			datasetToCreate.setOriginMetadata(originMetadata);
+			datasetToCreate.setUpdatedMetadata(originMetadata);
+			datasetToCreate.setBidsDataType(bidsDataType.getFolderName());
+
+			datasets.add(datasetToCreate);
+		}
+
+		datasetAcquisition.setDatasets(datasets);
+		datasetAcquisition.setAcquisitionEquipmentId(1L);
```

Comment:
Hi Jean-Come, I think the missing acquisition equipment is quite an issue, sorry, I was not seeing it before.
I would say the user should have to select it during the import OR we read it from, see BIDS spec, if possible:

Magnetic Resonance Imaging
Common metadata fields
MR Data described in the following sections share the following RECOMMENDED metadata fields (stored in sidecar JSON files). MRI acquisition parameters are divided
into several categories based on ”A checklist for fMRI acquisition methods reporting in the literature” by Ben Inglis:

Scanner Hardware
Key name Requirement level Data type Description
**Manufacturer** RECOMMENDED string Manufacturer of the equipment that
produced the composite instances.
Corresponds to DICOM Tag 0008, 0070
Manufacturer

**ManufacturersModelName** RECOMMENDED string Manufacturer’s model name of the
equipment that produced the composite
instances. Corresponds to DICOM Tag
0008, 1090 Manufacturers Model Name

**DeviceSerialNumber** RECOMMENDED string The serial number of the equipment
that produced the composite instances.
Corresponds to DICOM Tag 0018, 1000
DeviceSerialNumber. A pseudonym
can also be used to prevent the
equipment from being identifiable, so
long as each pseudonym is unique
within the dataset

Reply:
We have most of the equipments already in our database, I would guess